cmd/operator-sdk/bundle/create.go

```diff
@@ -96,7 +96,7 @@ $ operator-sdk bundle create \
 				defer cleanup()
```

Comment:
These cleanup funcs assume the generated files are present in `c.directory`. You'll have to set the correct cleanup dir in `c.cleanupFuncs()`:
```go
func (c bundleCmd) cleanupFuncs() (fs []func()) {
	dir := c.outputDir
	if dir == "" {
		dir = c.directory
	}
	metaDir := filepath.Join(dir, bundle.MetadataDir)
	dockerFile := filepath.Join(dir, bundle.DockerFile)
	...
}
```

Reply:
@estroz ah. so, the explicit purpose of --output-dir is to end up with a generated directory as output so that you don't have to keep generating them everytime you build. `annotations.yaml` is sort of supposed to be a long lived file that you can edit yourself manually.

It seems like `--generate-only` and `--output-dir` being set are either interchangeable (but with a specific dir set) OR passing the output-dir while also building would result in the files living around after you build.

IMO though it seems weird to me to generate and throw these files away all the time, given that OLM's packaging format is at some point probably going to want you to drive other stuff in the `metadata/` folder (ex. https://github.com/operator-framework/enhancements/blob/master/enhancements/operator-dependency-resolution.md#dependency-specification). I had assumed that we treated the bundle command as a "run once to scaffold things" rather than something that should default to deleting stuff afterwards. Maybe we should sync up on this?

Reply:
Also this change renames the `index.Dockerfile` and throws it in the same working directory as the context of where you're running the command. so if we end up wanting to clean it up, we will need to delete that file too.